### PR TITLE
sign release artifacts using cosign

### LINKED ISSUE #827: Sign `kapp` binaries while releasing them
**Describe the problem/challenge you have**
All published `kapp` binaries should be signed so that it can be verified that they were published through official channels.

**Describe the solution you'd like**
The changes proposed in [this proposal](https://github.com/carvel-dev/carvel/tree/develop/proposals/carvel/002-artifact-signatures) for signing binary artifacts must be made a part of the release pipeline.

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by prashantrewar**
#### Which issue(s) this PR fixes:

Fixes #827

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by rcmadhankumar** [comment]
@prashantrewar , Have you made any downstream releases to verify this workflow?

**Comment by prashantrewar** [comment]
hey @rcmadhankumar, yes i made downstream releases to verify this workflow, here is the [release](https://github.com/prashantrewar/kapp/releases/tag/v0.4.0) and [workflow](https://github.com/prashantrewar/kapp/actions/runs/7331915791).

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -86,6 +91,14 @@ jobs:
             ${checksums['kapp-linux-arm64']}  ./kapp-linux-arm64
             ${checksums['kapp-windows-amd64.exe']}  ./kapp-windows-amd64.exe`
 
+      - name: Verify checksums signature
+        run: |
+          cosign verify-blob \
+            --cert dist/checksums.txt.pem \
+            --signature dist/checksums.txt.sig \
+            --certificate-identity-regexp=https://github.com/carvel-dev \
```
Comment (rcmadhankumar) [inline .github/workflows/release.yml]: Shall we make this more generic?
```
--certificate-identity-regexp=https://github.com/${{ github.repository_owner }}
```

Comment (rcmadhankumar) [reply .github/workflows/release.yml]: cc: @praveenrewar  @100mik

Comment (prashantrewar) [reply .github/workflows/release.yml]: done!

**Comment by rcmadhankumar** [comment]
LGTM

**Comment by kumaritanushree** [comment]
LGTM

**Review by praveenrewar — APPROVED** [review]

**Pull request merged**